tests/test_api.py

```diff
@@ -128,6 +128,24 @@ def test_demo_streaming(test_client: TestClient[Litestar], api_key: str) -> None
         assert demo_in == demo_out
 
 
+def test_demo_streaming_no_late(test_client: TestClient[Litestar], api_key: str) -> None:
+    """Test streaming a demo to the database without a header overwrite."""
+    session_id = _open_mock_session(test_client, api_key).json()["session_id"]
+    assert isinstance(session_id, int)
+
+    _send_demo_file(test_client, api_key, str(session_id))
+
+    close_session_response = test_client.get("/close_session", params={"api_key": api_key})
+    assert close_session_response.status_code == HTTP_200_OK
+
+    with test_client.stream("GET", "/demodata", params={"api_key": api_key, "session_id": session_id}) as demo_stream:
+        demo_out = demo_stream.read()
+
+    with open("tests/data/test_demo.dem", "rb") as f:
+        demo_in = f.read()
+        assert demo_in == demo_out
```

Comment:
you could try out pytest.mark.parametrize here and paramaterize the late bytes or even just the params field and not duplicate code here

Reply:
v/r, i'm not gonna do that, feel free to check out the code and contribute it yourself